components/bitcoin-topics/markdown-ui/opcode-explorer.tsx

```diff
@@ -0,0 +1,449 @@
+        svgId: "ePRvDAyVHFs1"
+    },
+    OP_DUP: {
+        hex: "0x76",
+        description:
+            "Duplicates the top item on the stack and pushes it onto the stack again.",
+        asm: "5 OP_DUP",
+        hexCode: "0x05 0x76",
+        svgPath: "/bitcoin-topics/static/images/topics/overview/OP_DUP.svg",
+        svgId: "eGP7zSnXZvb1"
+    },
+    OP_EQUAL: {
+        hex: "0x87",
+        description:
+            "Compares the top two items on the stack. If they are equal, it pushes 1 onto the stack; otherwise, it pushes 0.",
+        asm: "5 5 OP_EQUAL",
+        hexCode: "0x05 0x05 0x87",
+        svgPath: "/bitcoin-topics/static/images/topics/overview/OP_EQUAL.svg",
+        svgId: "enWCL7dSHsN1"
+    },
+    OP_HASH160: {
+        hex: "0xA9",
+        description:
+            "Hashes the top item on the stack twice: first with SHA-256 and then with RIPEMD-160.",
+        asm: "<data> OP_HASH160",
+        hexCode: "<data> 0xA9",
+        svgPath: "/bitcoin-topics/static/images/topics/overview/OP_HASH160.svg",
+        svgId: "enqhsG8NxoX1"
+    },
+    OP_EQUALVERIFY: {
+        hex: "0x88",
+        description:
+            "Compares the top two items on the stack for equality. If they are not equal, the script fails.",
+        asm: "5 5 OP_EQUAL",
+        hexCode: "0x05 0x05 0x87",
+        svgPath:
+            "/bitcoin-topics/static/images/topics/overview/OP_EQUALVERIFY.svg",
+        svgId: "eO6lxqRz2Gw1"
+    },
+    OP_IF: {
+        hex: "0x63",
+        description:
+            "Evaluates a condition. If the condition is true (non-zero), it executes the statements after OP_IF. Otherwise, it jumps to the code after OP_ELSE, if present, or OP_ENDIF.",
+        asm: "<condition> OP_IF <true branch code> OP_ELSE <false branch code> OP_ENDIF",
+        hexCode:
+            "<condition> 0x63 <true branch code> 0x67 <false branch code> 0x68",
+        svgPath: "/bitcoin-topics/static/images/topics/overview/OP_IF.svg",
+        svgId: "eXi1yJfSEOH1"
+    },
+    OP_ELSE: {
+        hex: "0x67",
+        description:
+            "Marks the beginning of the false branch of a conditional execution (used after OP_IF). The code after OP_ELSE is executed if the condition evaluated by OP_IF is false (zero).",
+        asm: "OP_IF <true_statements> OP_ELSE <false_statements> OP_ENDIF",
+        hexCode: "0x63 <true_statements> 0x67 <false_statements> 0x68",
+        svgPath: "/bitcoin-topics/static/images/topics/overview/OP_ELSE.svg",
+        svgId: "exNKX69acQq1"
+    }
+}
+interface SvgatorPlayer {
+    ready: (callback: () => void) => void
+    pause: () => void
+    seekTo: (time: number) => void
+    play: () => void
+    duration: number
+    currentTime: number
+}
+
+const OpCodeExplorer = () => {
+    const [selectedOpCode, setSelectedOpCode] = useState<string>(
+        Object.keys(opcodeData)[0]
+    )
+    const [isAsm, setIsAsm] = useState<boolean>(true)
```

Comment:
A general thing I noticed, when you use a useState and define it's initial value, the useState automatically defines it's type both for the state and the setState so there might be no need for defining its type in angle braces. 

eg: ``const [state, setState] = useState<boolean>(true)`` 

- the state already defines it's type from the initial state passed which is `true` to be a type of boolean
- the setState also defines it's type to be a type of  `React.Dispatch<React.SetStateAction<boolean>>`, notably a setStateAction of the type of <boolean>

Defining a state's type becomes required when you're dealing with complex data structures